src/mpyl/run_plan.py

```diff
@@ -1,38 +1,61 @@
 """This module contains the RunPlan class."""
-
 from dataclasses import dataclass
+from typing import Optional
 
 from .project import Project, Stage
 from .project_execution import ProjectExecution
 
 
-@dataclass(frozen=True)
+@dataclass()
 class RunPlan:
-    plan: dict[Stage, set[ProjectExecution]]
+    full_plan: dict[Stage, set[ProjectExecution]]
+    _selected_stages: Optional[Stage] = None
+    _selected_projects: Optional[set[Project]] = None
+    _selected_plan: Optional[dict[Stage, set[ProjectExecution]]] = None
 
     @staticmethod
     def empty() -> "RunPlan":
-        return RunPlan({})
+        return RunPlan(full_plan={})
 
-    def get(self, stage: Stage) -> set[ProjectExecution]:
-        return self.plan.get(stage, set())
+    @property
+    def selected_stage(self) -> Optional[Stage]:
+        return self._selected_stages
 
-    def add_stage(self, stage: Stage, executions: set[ProjectExecution]):
-        self.plan.update({stage: executions})
+    @selected_stage.setter
+    def selected_stage(self, stage: Stage):
+        self._selected_stages = stage
 
-    def update(self, run_plan: "RunPlan"):
-        self.plan.update(run_plan.plan)
+    @property
+    def selected_projects(self) -> Optional[set[Project]]:
+        return self._selected_projects
+
+    @selected_projects.setter
+    def selected_projects(self, projects: set[Project]):
+        self._selected_projects = projects
+
+    @property
+    def selected_plan(self) -> dict[Stage, set[ProjectExecution]]:
+        if not self.selected_stage and not self._selected_projects:
+            return self.full_plan
 
-    def items(self):
-        return self.plan.items()
+        self._selected_plan = {}
```

Comment:
Don't think this should be a class variable. I think you can just do `selected_plan = {}`.